### PR TITLE
Fix incorrect error codes, and add better messaging

### PR DESCRIPTION
**Pull request by sdford**
* Return the proper error codes and messages for:
    * Bad username, password or MFA
    * Invalid region during IAM authentication (Issue #18)
    * Duplicate SDB names

### CODE REVIEW

File: src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java [hunk 1]
```diff
@@ -157,7 +157,7 @@ protected SessionLoginTokenData verifyFactor(final String deviceId,
             String msg = String.format("stateToken: %s failed to verify 2nd factor for reason: %s",
                     stateToken, verifyFactorResponse.getStatus().getMessage());
 
-            if (verifyFactorResponse.getStatus().getCode() == 400L) {
+            if (verifyFactorResponse.getStatus().getCode() == 401) {
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java]: You are introducing a bug here.

This code is verifying the response code from the OneLogin API call, which would still be a `400`, unless I am mistaken.

File: src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java [hunk 2]
```diff
@@ -184,20 +184,21 @@ protected SessionLoginTokenData createSessionLoginToken(final String username, f
...
-                            .withApiErrors(DefaultApiError.MFA_SETUP_REQUIRED)
-                            .withExceptionMessage(msg)
-                            .build();
-                } else {
-                    throw ApiException.newBuilder()
-                            .withApiErrors(DefaultApiError.AUTH_BAD_CREDENTIALS)
-                            .withExceptionMessage(msg)
-                            .build();
-                }
+            if (statusCode == 400 &&
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java]: Again, unless OneLogin changed there API, this code is going to be introducing bugs no?

File: src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java [hunk 3]
```diff
@@ -184,20 +184,21 @@ protected SessionLoginTokenData createSessionLoginToken(final String username, f
...
+                    StringUtils.startsWithIgnoreCase(createSessionLoginTokenResponse.getStatus().getMessage(), "MFA")) {
+                throw ApiException.newBuilder()
+                        .withApiErrors(DefaultApiError.MFA_SETUP_REQUIRED)
+                        .withExceptionMessage(msg)
+                        .build();
+            } else if (statusCode == 401) {
+                throw ApiException.newBuilder()
+                        .withApiErrors(DefaultApiError.AUTH_BAD_CREDENTIALS)
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java]: In DefaultApiError the status code for this needs to change from 400 -> 401

File: src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java [hunk 4]
```diff
@@ -184,20 +184,21 @@ protected SessionLoginTokenData createSessionLoginToken(final String username, f
...
+                        .withExceptionMessage(msg)
+                        .build();
             } else {
                 throw ApiException.newBuilder()
                         .withApiErrors(DefaultApiError.GENERIC_BAD_REQUEST)
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java]: This is also where that `The request will not be completed` message is getting generated from and needs to be cleaned up with a better message and a `401`

File: src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java (hunk 1 above)
Comment (sdford) [inline src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java]: Unfortunately, OneLogin changed what they're returning here without changing the API version.

File: src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java (hunk 3 above)
Comment (sdford) [inline src/main/java/com/nike/cerberus/auth/connector/onelogin/OneLoginAuthConnector.java]: Makes sense. I will fix that as a part of this PR too.

**Pull request merged**